controlplane/internal/imageregistry/image_registry_test.go

```diff
@@ -232,57 +202,108 @@ var _ = Describe("ImageRegistry Test", func() {
 
 		When("the user-provided image registry ConfigMap is missing the registries.conf key", func() {
 			It("fails to create the image registry configmap", func() {
-				userRegistryCM := generateUserProvidedRegistryCM(
-					providedCMName,
+				userRegistryCM := newUserProvidedRegistryCM("", certificate)
+
+				By("Calling the CreateImageRegistryConfigmap function")
+				imageRegistryConfigMap, err := CreateImageRegistryConfigmap(
+					userRegistryCM,
 					namespace,
-					"",
-					certificate,
 				)
+				Expect(err).To(HaveOccurred())
+				Expect(imageRegistryConfigMap).To(BeNil())
+			})
+		})
 
+		When("the user-provided image registry ConfigMap has an invalid toml configuration", func() {
+			It("returns parsing error and fails to create configmap", func() {
+				userRegistryCM := newUserProvidedRegistryCM(getInvalidRegistryToml(), certificate)
 				By("Calling the CreateImageRegistryConfigmap function")
 				imageRegistryConfigMap, err := CreateImageRegistryConfigmap(
 					userRegistryCM,
 					namespace,
 				)
 				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(Equal("(6, 13): cannot have two dots in one float"))
 				Expect(imageRegistryConfigMap).To(BeNil())
 			})
 		})
+
 	})
 })
 
-func getRegistryToml(source, mirror string, insecure, pullByTag bool) string {
-	registryToml := ""
-	if source != "" {
-		registryToml = fmt.Sprintf("[[registry]]\nlocation = \"%s\"", source)
-	}
-	if mirror != "" {
-		registryToml = fmt.Sprintf("%s\n[[registry.mirror]]\nlocation = \"%s\"", registryToml, mirror)
-	}
-	if insecure {
-		registryToml = fmt.Sprintf("%s\ninsecure = true", registryToml)
-	}
-	if pullByTag {
-		registryToml = fmt.Sprintf("%s\npull-from-mirror = \"tag-only\"", registryToml)
-	}
-	return registryToml
+func getSecureRegistryToml() string {
```

Comment:
@CrystalChun I think trying to use code as simple as possible in the tests always helps. Getting rid of conditions makes clearer what we are testing, and reduce probability of errors in the test itself, otherwise we'd need to write a test for the test :)